### PR TITLE
Fix utils coding style

### PR DESCRIPTION
**Pull request by ender503**
Fix utils coding style

### CODE REVIEW

**Comment by ender503** [comment]
@HexRabbit I need your help to review my changes, thanks! 👍

File: utils/combineSortedArray.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const deepCopy = require('./deepCopy');
```
Comment (jserv) [inline utils/combineSortedArray.js]: The file was added specifically to this commit. Can you explain?

**Comment by jserv** [comment]
The subject might be misleading since @ender503 changed a lot and even add new file. It is not straightforward for @HexRabbit to review. Thus, I would suggest to improve Git commit messages to address certain non-style changes.

File: utils/combineSortedArray.js (hunk 1 above)
Comment (ender503) [reply utils/combineSortedArray.js]: The `combineSortedArray` function was defined in the `mamUtils.js`, it looks like a generic utility function so I move it out of the `mamUtils.js`.

File: utils/localStore.js [hunk 2]
```diff
@@ -1,177 +1,174 @@
...
+      if (self.messages === null) {
+        self.messages = self.db.addCollection('messages');
       }
-      callback()
-    }
+      callback();
+    };
 
-    this.db = new loki(dbFileName, {
+    this.db = new Loki(dbFileName, {
```
Comment (jserv) [inline utils/localStore.js]: Why did you change from lowercase `loki` to uppercase `Loki`?

File: utils/tools.js [hunk 3]
```diff
@@ -1,116 +1,72 @@
...
-};
+const signHash = (msg, sk) => sha256(e64(crypto.privateEncrypt(eUTF(d64(sk)), Buffer.from(dUTF(msg))))); // eslint-disable-line max-len
 // Encrypt data for transmit
-const encrypt = (msg, pk) => {
-  return e64(crypto.publicEncrypt(eUTF(d64(pk)), Buffer(dUTF(msg))));
-};
+const encrypt = (msg, pk) => e64(crypto.publicEncrypt(eUTF(d64(pk)), Buffer.from(dUTF(msg))));
 
 // Decrypt data
-const decryptUTF = (msg, sk) => {
```
Comment (jserv) [inline utils/tools.js]: To me, it is NOT trivial to interpret these lines as "style tweaks".

File: utils/combineSortedArray.js (hunk 1 above)
Comment (HexRabbit) [reply utils/combineSortedArray.js]: The `combineSortedArray` function may not be able to use in other situation. So I think just keep it in `mamUtils.js` will be nice.

File: utils/tools.js (hunk 3 above)
Comment (ender503) [reply utils/tools.js]: This is Airbnb coding style use arrow-body-style rule with `as-needed` option, so I change these line to fit the coding style.

If it is inappropriately, maybe we could use this rule with `always`. 🤔 

See the different in this [link](https://eslint.org/docs/rules/arrow-body-style).

Comment (jserv) [reply utils/tools.js]: I defer to `STYLE-GUIDE.md` and you shall ensure most style tweaks can be automated without human interactions.

File: utils/localStore.js (hunk 2 above)
Comment (ender503) [reply utils/localStore.js]: To me, it is helpful to use uppercase to recognize which is constructor function. 🤔

File: utils/tools.js (hunk 3 above)
Comment (ender503) [reply utils/tools.js]: This coding style can be automatically fixed by the `eslint -fix`.
[List of available rules](https://eslint.org/docs/rules/)

**Comment by ender503** [comment]
@jserv I think this PR could be closed, I'll submit another PR to split commit into smaller.
And that PR would be easy to be reviewed.

**Comment by jserv** [comment]
You have to improve Git commit messages to reflect exact changes.

File: utils/localStore.js (hunk 2 above)
Comment (jserv) [reply utils/localStore.js]: Create a new pull request to summarize key principles of style we would like to enforce in file `STYLE-GUIDE.md`. Otherwise, other developers are unlikely performing well.

Comment (jserv) [reply utils/localStore.js]: More examples illustrating coding style would be necessary though.

**Comment by jserv** [comment]
Let's close this issue and concentrate on documentation about further style manner.
